tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -799,6 +799,68 @@ public:
    void CommitCluster() final;
 };
 
+/// The field for values that may or may not be present in an entry. Parent class for unique pointer field and
+/// optional field. A nullable field cannot be instantiated itself but only its descendants.
+/// The RNullableField takes care of the on-disk representation. Child classes are charged with the in-memory
+/// representation.  The on-disk representation can be "dense" or "sparse". Dense nullable fields have a bitmask
+/// (true: item available, false: item missing) and serialize a default-constructed item for missing items.
+/// Sparse nullable fields use a switch column to point to the available items.
+/// By default, items whose size is smaller or equal to 8 bytes (size of switch column element) are stored densely.
+class RNullableField : public Detail::RFieldBase {
+   /// For a dense nullable field, used to write a default-constructed item for missing ones.
+   Detail::RFieldValue fDefaultItemValue;
+   /// For a sparse nullable field, the number of written non-null items in this cluster
+   ClusterSize_t fNWritten{0};
+
+protected:
+   const Detail::RFieldBase::RColumnRepresentations &GetColumnRepresentations() const final;
+   void GenerateColumnsImpl() final;
+   void GenerateColumnsImpl(const RNTupleDescriptor &) final;
+
+   std::size_t AppendNull();
+   std::size_t AppendValue(const Detail::RFieldValue &value);
+   /// Given the index of the nullable field, returns the corresponding global index of the subfield or,
+   /// if it is null, returns kInvalidClusterIndex
+   RClusterIndex GetItemIndex(NTupleSize_t globalIndex);
+
+   RNullableField(std::string_view fieldName, std::string_view typeName,
+                  std::unique_ptr<Detail::RFieldBase> &&itemField);
```

Comment:
`std::unique_ptr` cannot be copied anyway; to be consistent with other uses in RNTuple (in this and other files), I would suggest using the form
```suggestion
                  std::unique_ptr<Detail::RFieldBase> itemField);
```